test/core/client_channel/load_balanced_call_destination_test.cc

```diff
@@ -196,6 +195,36 @@ LOAD_BALANCED_CALL_DESTINATION_TEST(StartCall) {
   WaitForAllPendingWork();
 }
 
+LOAD_BALANCED_CALL_DESTINATION_TEST(StartCallOnDestroyedChannel) {
```

Comment:
As per our discussion, it's not entirely clear to me which case this is testing.  The test name implies that it's testing a destroyed channel, but if I'm understanding the test itself, it's actually covering a case where the picker is not yet set, and I don't think that can actually happen in practice.

Reply:
added some comments around this